crates/rome_js_formatter/src/utils/mod.rs

```diff
@@ -649,3 +650,109 @@ pub(crate) fn is_call_like_expression(expression: &JsAnyExpression) -> bool {
             | JsAnyExpression::JsCallExpression(_)
     )
 }
+
+/// Data structure used to merge into one the following nodes:
+///
+/// - [JsAnyObjectMemberName]
+/// - [JsAnyClassMemberName]
+/// - [JsLiteralMemberName]
+///
+/// Once merged, the enum is used to get specific members (the literal ones) and elide
+/// the quotes from them, when the algorithm sees fit
+pub(crate) enum PropertyName {
+    Object(JsAnyObjectMemberName),
+    Class(JsAnyClassMemberName),
+    Literal(JsLiteralMemberName),
+}
+
+impl From<JsAnyClassMemberName> for PropertyName {
+    fn from(node: JsAnyClassMemberName) -> Self {
+        Self::Class(node)
+    }
+}
+
+impl From<JsAnyObjectMemberName> for PropertyName {
+    fn from(node: JsAnyObjectMemberName) -> Self {
+        Self::Object(node)
+    }
+}
+
+impl From<JsLiteralMemberName> for PropertyName {
+    fn from(literal: JsLiteralMemberName) -> Self {
+        Self::Literal(literal)
+    }
+}
+
+const QUOTES_TO_OMIT: [char; 2] = ['\"', '\''];
+
+pub(crate) enum PropertyNameCheckMode {
+    Alphabetic,
+    Alphanumeric,
+}
+
+impl PropertyNameCheckMode {
+    /// We can change the text only if there alphanumeric or alphabetic characters, depending on the mode
+    fn text_can_be_replaced(&self, text_to_check: &str) -> bool {
+        match self {
+            PropertyNameCheckMode::Alphabetic => text_to_check.chars().all(char::is_alphabetic),
+            PropertyNameCheckMode::Alphanumeric => text_to_check.chars().all(char::is_alphanumeric),
+        }
+    }
+}
+
+/// Function used by the formatter, where we pass a complaint member and it returns a [FormatElement[
+/// where the text has its quotes removed.
+pub(crate) fn format_property_name<Member: Into<PropertyName>>(
+    member_name: Member,
+    formatter: &Formatter,
+    checker: PropertyNameCheckMode,
+) -> FormatResult<FormatElement> {
+    fn replace_node(
+        name: JsSyntaxToken,
+        formatter: &Formatter,
+        checker: PropertyNameCheckMode,
+    ) -> FormatResult<FormatElement> {
+        let text = name.text_trimmed();
+
+        // there are cases where we might have an empty string, which means that if the text length
+        // is higher than 2, it means that the text has something in it
```

Comment:
```suggestion
        // is higher than 2, it means that the identifier name isn't empty
```

It might be worth to move this test into the `text_can_be_replaced`. It then becomes a simple `ident.is_empty()`.